draft-ietf-sframe-enc.md

```diff
@@ -606,20 +606,43 @@ the other participants.
 
 In this scheme, it is assumed that receivers have a signal outside of SFrame for
 which client has sent a given frame (e.g., an RTP SSRC).  SFrame KID
-values are then used to distinguish generations of the sender's key.  At the
-beginning of a call, each sender encrypts with KID=0.  Thereafter, the sender
-can ratchet their key forward for forward secrecy:
+values are then used to distinguish versions of the sender's key.
 
+Key IDs in this scheme have two parts, a "key generation" and a "ratchet step".
+Both are unsigned integers that begin at zero.  The key generation increments
+each time the sender distributes a new key to receivers.  The "ratchet step" is
+increment each time the sender ratchets their key forward for forward secrecy:
+
+~~~~~ pseudocode
+sender_base_key[i+1] = HKDF-Expand(
+                         HKDF-Extract(sender_base_key[i], 'SFrame 1.0 ratchet'),
+                         '', CipherSuite.Nh)
 ~~~~~
-sender_key[i+1] = HKDF-Expand(
-                    HKDF-Extract(sender_key[i], 'SFrame 1.0 ratchet'),
-                      '', AEAD.Nk)
+
+For compactness, we do not send the whole epoch number.  Instead, we send only
+its low-order `R` bits, where `R` is a value set by the application.  `R`
```

Comment:
should we add a note saying that R needs to be exchanged between the sender and receiver by out of band mechanism ?